Index: src/com/example/hospital/HealthProfile.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/com/example/hospital/HealthProfile.java	(date 1591785178922)
+++ src/com/example/hospital/HealthProfile.java	(date 1591785178922)
@@ -14,7 +14,7 @@
     private HeartRates heartRates;
     private int weight;
     private int height;
-    private BMICalculator bmiCalculator;
+    private final BMICalculator bmiCalculator;
     private String condition;
     private int choice;
 
@@ -32,6 +32,7 @@
         if (dateOfBirth != null) {
             heartRates = new HeartRates(dateOfBirth);
         }
+        bmiCalculator = new BMICalculator(weight, height);
     }
 
     public String getFirstName() {
@@ -119,7 +120,6 @@
     }
 
     private double bmiRate(int choice) {
-        bmiCalculator = new BMICalculator(getWeight(), getHeight());
        if (choice == 1){
            return bmiCalculator.bmiAmerican(getWeight(), getHeight());
        }
